src/mlpack/methods/lmnn/lmnn_main.cpp

```diff
@@ -0,0 +1,350 @@
+    "with " + PRINT_PARAM_STRING("batch_delta") + ").  In addition, "
+    "a normalized starting point can be used by specifying the " +
+    PRINT_PARAM_STRING("normalize") + " parameter. "
+    "\n\n"
+    "Stochastic gradient descent, specified by the value 'sgd' for the "
+    "parameter " + PRINT_PARAM_STRING("optimizer") + ", depends "
+    "primarily on three parameters: the step size (specified with " +
+    PRINT_PARAM_STRING("step_size") + "), the batch size (specified with " +
+    PRINT_PARAM_STRING("batch_size") + "), and the maximum number of passes "
+    "(specified with " + PRINT_PARAM_STRING("passes") + ").  In "
+    "addition, a normalized starting point can be used by specifying the " +
+    PRINT_PARAM_STRING("normalize") + " parameter. "
+    "\n\n"
+    "The L-BFGS optimizer, specified by the value 'lbfgs' for the parameter " +
+    PRINT_PARAM_STRING("optimizer") + ", uses a back-tracking line search "
+    "algorithm to minimize a function.  The following parameters are used by "
+    "L-BFGS: " + PRINT_PARAM_STRING("num_basis") + " (specifies the number"
+    " of memory points used by L-BFGS), " +
+    PRINT_PARAM_STRING("max_iterations") + ", " +
+    PRINT_PARAM_STRING("armijo_constant") + ", " +
+    PRINT_PARAM_STRING("wolfe") + ", " + PRINT_PARAM_STRING("tolerance") +
+    " (the optimization is terminated when the gradient norm is below this "
+    "value), " + PRINT_PARAM_STRING("max_line_search_trials") + ", " +
+    PRINT_PARAM_STRING("min_step") + ", and " +
+    PRINT_PARAM_STRING("max_step") + " (which both refer to the line search "
+    "routine).  For more details on the L-BFGS optimizer, consult either the "
+    "mlpack L-BFGS documentation (in lbfgs.hpp) or the vast set of published "
+    "literature on L-BFGS."
+    "\n\n"
+    "By default, the SGD optimizer is used.");
+
+PARAM_MATRIX_IN_REQ("input", "Input dataset to run LMNN on.", "i");
+PARAM_UROW_IN("labels", "Labels for input dataset.", "l");
+PARAM_INT_IN("num_targets", "Number of target neighbors to use for each "
+    "datapoint.", "k", 1);
+PARAM_MATRIX_OUT("output", "Output matrix for learned distance matrix.", "o");
+PARAM_STRING_IN("optimizer", "Optimizer to use; 'amsgrad', 'bbsgd', 'sgd', or "
+    "'lbfgs'.", "O", "amsgrad");
+PARAM_DOUBLE_IN("regularization", "Regularization for LMNN objective function ",
+    "r", 0.5);
+PARAM_FLAG("normalize", "Use a normalized starting point for optimization. It"
+    "is useful for when points are far apart, or when SGD is returning NaN.",
+    "N");
+PARAM_INT_IN("passes", "Maximum number of full passes over dataset for "
+    "AMSGrad, BB_SGD and SGD.", "p", 50);
+PARAM_INT_IN("max_iterations", "Maximum number of iterations for "
+    "L-BFGS (0 indicates no limit).", "n", 100000);
+PARAM_DOUBLE_IN("tolerance", "Maximum tolerance for termination of AMSGrad, "
+    "BB_SGD, SGD or L-BFGS.", "t", 1e-7);
+PARAM_DOUBLE_IN("step_size", "Step size for AMSGrad, BB_SGD and SGD."
```

Comment:
I think the `.` should be removed here and replaced with a space, so the `(alpha).` is included in the same sentence.

Reply:
Done!